src/static/js/html10n.js

```diff
@@ -181,16 +181,20 @@ window.html10n = (function(window, document, undefined) {
 
     // dat alng ain't here, man!
     if (!data[lang]) {
-      var msg = 'Couldn\'t find translations for '+lang
-        , l
-      if(~lang.indexOf('-')) lang = lang.split('-')[0] // then let's try related langs
-      for(l in data) {
-        if(lang != l && l.indexOf(lang) === 0 && data[l]) {
-          lang = l
-          break;
+      if (lang.indexOf('-') > -1) {
+        var msg = 'Couldn\'t find translations for '+lang
+        lang = lang.split('-')[0] // get root locale incase variant (e.g ru-RU)
+        if (!data[lang]) { // check root locale (e.g ru)
+          var l
+          for(l in data) { // try similar (e.g zh-hans)
+            if(lang != l && l.indexOf(lang) === 0 && data[l]) {
+              lang = l
+              break;
+            }
+          }
+          if(lang != l) return cb(new Error(msg))
         }
       }
-      if(lang != l) return cb(new Error(msg))
```

Comment:
This code is clever.

Could you add a description _in natural language_ of what this section is doing?
An example would be ok.

Imagine to have a nitwit in front of you.

Reply:
Also, the `var` is a bit flinky in such deep scopes in js, but in this case we can live without `let`. Just saying.